testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionUpdateProfileTest.java

```diff
@@ -258,7 +262,10 @@ public void updateProfileMissingEmail() {
         Assert.assertEquals("New last", updateProfilePage.getLastName());
         Assert.assertEquals("", updateProfilePage.getEmail());
 
-        Assert.assertEquals("Please specify email.", updateProfilePage.getInputErrors().getEmailError());
+        assertThat(updateProfilePage.getInputErrors().getEmailError(), anyOf(
```

Comment:
Isn't `anyOf` a bit too nondeterministic for a test? :) But I'm probably just missing some context why we need to check for both error messages.

Reply:
Legacy and declarative user profile have different messages.